neqo-http3/src/connection_server.rs

```diff
@@ -64,13 +64,15 @@ impl Http3ServerHandler {
         data: &[u8],
         conn: &mut Connection,
     ) -> Res<usize> {
-        self.base_handler.stream_has_pending_data(stream_id);
-        self.needs_processing = true;
-        self.base_handler
+        let n = self
+            .base_handler
             .send_streams
             .get_mut(&stream_id)
             .ok_or(Error::InvalidStreamId)?
-            .send_data(conn, data)
+            .send_data(conn, data)?;
+        self.base_handler.stream_might_have_pending_data(stream_id);
```

Comment:
The reordering here is good, but there might be a simpler solution:

```suggestion
        if n > 0 {
            self.base_handler.stream_might_have_pending_data(stream_id);
        }
```

I looked and the other two call sites are either OK or can be addressed differently.